drivers/auth/auth_mod.c

```diff
@@ -59,9 +58,9 @@ extern unsigned int auth_img_flags[];
 static int cmp_auth_param_type_desc(const auth_param_type_desc_t *a,
 		const auth_param_type_desc_t *b)
 {
-	if ((a->type == b->type) && (a->cookie == b->cookie)) {
+	if (a->type == b->type && a->cookie == b->cookie)
```

Comment:
I do not like your idea, but I am following you
just because I do not want this pull request blocked.